### ISSUE TITLE
How to differentiate bringing the app to the foreground via a notification tap

### ISSUE THREAD

**Issue by pakoArtal** [morinel/gcmpush#39]
Hi,
This is be related to  the issue [App open from PushNotification click without payload](https://github.com/morinel/gcmpush/issues/26) where @cjadhav and @DFoxinator ask about getting the payload on clicking on notification.

To differentiate bringing the app to the foreground via a notification tap from just bringing the app to the foreground normally, I have tried the following:

-  At GCMIntentService.java add an Extra to the launcherIntent :  
```java
launcherIntent.putExtra("PUSH_LAUNCH", true);`
```
-  Somewhere in my code add an '[newIntent](http://docs.appcelerator.com/platform/latest/#!/api/Titanium.Android.Activity-event-newintent)' event listener. 
I have tried with no luck: 
```javascript
win.activity.addEventListener("newIntent", function(e) {
     console.log('NEW INTENT ' + JSON.stringify(e.intent) + ' WITH EXTRA: ' + e.intent.getBooleanExtra('PUSH_LAUNCH', false));
});
```
and 
```javascript
Ti.Android.currentActivity.addEventListener('newintent', function(e) {
		 console.log('NEW INTENT ' + JSON.stringify(e.intent) + ' WITH EXTRA: ' + e.intent.getBooleanExtra('PUSH_LAUNCH', false));
});
```
@FokkeZB opened a JIRA ticket :[Android newintent event not fired](https://jira.appcelerator.org/browse/TIMOB-20459) so that could be reason I do not get it. 

Do you know another way to do it?

**Comment by DFoxinator** [comment]
@pakoArtal were you ever able to figure this out?

**Comment by pakoArtal** [comment]
Hi @DFoxinator,
No luck yet, if I get news I will write them here...
If I understand correctly we need  a 'newIntent' listener.  In the JIRA ticket related to the newIntent listener says:
Resolution: Unresolved
Affects Version/s: Release 5.2.0
Fix Version/s: Release **6.0.0**

To get notified, Login to jira and press 'Start watching this issue' on https://jira.appcelerator.org/browse/TIMOB-4865

I heard that the more people watching the issue the fastest they solve it ....

**Comment by DFoxinator** [comment]
@pakoArtal thanks a bunch, what you're saying makes sense. The way you were trying to do it with newintent listener is what I was trying to do too.

I left a comment on the ticket asking about a workaround. This seems like very critical functionality, if in fact you literally can't properly open a Titanium app that's in the background from a push notification with a payload.

**Comment by bitfabrikken** [comment]
You can do it yourself by using gcm.getLastData(); when you start the app

**Comment by mark22** [comment]
Agreed, this is critical functionality.  Surprised this isn't included somehow??
